File: 3487. Maximum Unique Subarray Sum After Deletion.java
class Solution {
    public int maxSum(int[] nums) {
        HashSet<Integer> hs = new HashSet<>();
        int n = nums.length;

        int sum = 0;
        for(int num: nums){
            if(!hs.contains(num) && num > 0){
                hs.add(num);
                sum += num;
            }
        }
        if(sum == 0){
            sum = nums[0];
            for(int num: nums){
                if(sum < num) sum = num;
            }
        }
        return sum;
    }
}

File: 3487. Maximum Unique Subarray Sum After Deletion.cpp
class Solution {
public:
    int maxSum(vector<int>& nums) {
        unordered_set<int> seen;
        int sum = 0;
        bool hasPositive = false;

        for (int num : nums) {
            if (num > 0 && seen.find(num) == seen.end()) {
                seen.insert(num);
                sum += num;
                hasPositive = true;
            }
        }

        if (!hasPositive) {
            sum = nums[0];
            for (int num : nums) {
                sum = max(sum, num);
            }
        }

        return sum;
    }
};
